### PR TITLE
Problem: Omnigres docker image doesn't pick up current omni version

### PR DESCRIPTION
**Pull request by rmodpur**
Omni version is hardcoded in docker-entrypoint-initdb.d sql file
containing `shared_preload_libraries`. On release of new omni
version the container doesn't start because it points to older
omni version which is not present in the container image.

Solution: convert docker-entrypoint-initdb.d scripts to templates
and substitute the env variables used in it using `envsubst`

### CODE REVIEW

**Comment by rmodpur** [comment]
Is `RelWithDebInfo` the right default for docker image because we release extensions for `Release` build type ?
https://github.com/rmodpur/omnigres/blob/docker-omni-version/Dockerfile#L6

**Comment by yrashk** [comment]
> Is `RelWithDebInfo` the right default for docker image because we release extensions for `Release` build type ? https://github.com/rmodpur/omnigres/blob/docker-omni-version/Dockerfile#L6

I think the original reason for this was that containers are primarily meant for development/testing, not production.

File: Dockerfile [hunk 1]
```diff
@@ -94,7 +94,20 @@ ENV POSTGRES_PASSWORD=omnigres
 COPY --from=build /build/packaged /omni
 COPY --from=build /build/python-index /python-packages
 COPY --from=build /build/python-wheels /python-wheels
-COPY docker/initdb-slim/* /docker-entrypoint-initdb.d/
+# need versions.txt to pick omni version
+COPY --from=build /omni/versions.txt /omni/versions.txt
+RUN cat <<EOF > /docker-entrypoint-initdb.d/001-settings.sql
+alter system set max_worker_processes = 64;
```
[the remark this answers is not in the document]
Comment (rmodpur) [reply Dockerfile:100]: noted, will use templates

**Comment by yrashk** [comment]
Just thinking out loud: do you anticipate any gotchas from using shell-style variables in SQL script templates?

**Comment by rmodpur** [comment]
> Just thinking out loud: do you anticipate any gotchas from using shell-style variables in SQL script templates?

Given we control the values we are injecting and only substituting [specific env variables](https://www.baeldung.com/linux/envsubst-command#specify-variables-with-shell-format), I don't see any issue. The variable substitution is only one time thing, the templates files are removed and converted to sql and shell scripts in dockerfile itself.

**Review by yrashk — APPROVED** [review]

**Pull request merged**